test/SwapIntentInvertBid.t.sol

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.22;
+
+import "forge-std/Test.sol";
+
+import { ERC20 } from "solmate/tokens/ERC20.sol";
+
+import { BaseTest } from "./base/BaseTest.t.sol";
+import { TxBuilder } from "src/contracts/helpers/TxBuilder.sol";
+
+import { SolverOperation } from "src/contracts/types/SolverCallTypes.sol";
+import { UserOperation } from "src/contracts/types/UserCallTypes.sol";
+import { DAppOperation, DAppConfig } from "src/contracts/types/DAppApprovalTypes.sol";
+
+import { SwapIntent, SwapIntentInvertBidDAppControl } from "src/contracts/examples/intents-example/SwapIntentInvertBidDAppControl.sol";
+import { SolverBaseInvertBid } from "src/contracts/solver/SolverBaseInvertBid.sol";
+
+interface IUniV2Router02 {
+    function swapExactTokensForTokens(
+        uint256 amountIn,
+        uint256 amountOutMin,
+        address[] calldata path,
+        address to,
+        uint256 deadline
+    )
+        external
+        returns (uint256[] memory amounts);
+}
+
+contract SwapIntentTest is BaseTest {
+    SwapIntentInvertBidDAppControl public swapIntentControl;
+    TxBuilder public txBuilder;
+    Sig public sig;
+
+    ERC20 DAI = ERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address DAI_ADDRESS = address(DAI);
+
+    struct Sig {
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+
+    function setUp() public virtual override {
+        BaseTest.setUp();
+
+        // Creating new gov address (SignatoryActive error if already registered with control)
+        governancePK = 11_112;
+        governanceEOA = vm.addr(governancePK);
+
+        // Deploy new SwapIntent Control from new gov and initialize in Atlas
+        vm.startPrank(governanceEOA);
+        swapIntentControl = new SwapIntentInvertBidDAppControl(address(atlas));
+        atlasVerification.initializeGovernance(address(swapIntentControl));
+        vm.stopPrank();
+
+        txBuilder = new TxBuilder({
+            _control: address(swapIntentControl),
+            _atlas: address(atlas),
+            _verification: address(atlasVerification)
+        });
+
+        // Deposit ETH from Searcher signer to pay for searcher's gas
+        // vm.prank(solverOneEOA);
+        // atlas.deposit{value: 1e18}();
+    }
+
+    function testAtlasSwapIntentInvertBidWithBasicRFQ() public {
```

Comment:
Nice! Can we do _OneSolver and _TwoSolver versions of this test where in Two we also use bidFind and the 2nd solver in the array has a lower bid, so we check that they win and the first solver in the array does not win?

I'm not actually sure if the _bidFindingIteration() function would be compatible with this. Pls investigate if it isnt

Reply:
Will approve after this test is added

Reply:
If invertBidValue is set, I don't think bidFind is possible as how will atlas know how much `bidToken` to send to the solver if it isn't pre-specified?

Reply:
OK I’m happy with that and I don’t think we need to support bidFind and invertsBidValue together.
I think we could support it by just transferring the users maxAmountOut to the solver in the preSolver hook, and specifying that solvers must return any surplus before the end of solverExrcution. That should let us do a bidFind for inverted bid, but like I said that’s more of a nice-to-have that I don’t think we need to support.